backend.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type Backend interface {
+	BackendReader
+	BackendWriter
+}
+
+// Implement to store event
+type BackendWriter interface {
+	insert(DomainEntry) error
+	ping(HostEntry) error
+}
+
+// Implement to retrieve event from backend and convert to QueryResponse
+type BackendReader interface {
+	get(string) (QueryResponse, error)
+	update() ([]HostEntry, error)
+}
+
+// distributedBackend
+type distributedBackend struct {
+	hosts       []string
+	persistance Backend
+}
+
+func NewDistributedBackend(
+	hosts []string, persistBackend Backend) *distributedBackend {
+	b := new(distributedBackend)
+	b.hosts = hosts
+	b.persistance = persistBackend
+	return b
+}
+
+func (b distributedBackend) ping(entry HostEntry) error {
+	return b.persistance.ping(entry)
+}
+
+func (b distributedBackend) get(domain string) (QueryResponse, error) {
+	var total int = 0
+	bounced := false
+	resp, err := b.persistance.get(domain)
+	if err != nil {
+		log.Fatalf("Error selecting from persistance: %s", err)
+	}
+	if resp.Bounced {
+		log.Println("Short circuited P2P due to persistance")
+		ret := QueryResponse{Bounced: true, Total: 0, Domain: domain}
+		return ret, nil
+	}
+	total += resp.Total
```

Comment:
It is possible to know if the domain is catch-all by this point and we could avoid the rest of this method.